include/ccf/historical_queries_adapter.h

```diff
@@ -83,9 +83,148 @@ namespace ccf::historical
     return true;
   };
 
+  enum class HistoricalTxStatus
+  {
+    Error,
+    PendingOrUnknown,
+    Invalid,
+    Valid
+  };
+
+  using CheckHistoricalTxStatus = std::function<HistoricalTxStatus(
+    ccf::View view, ccf::SeqNo seqno, std::string& error_reason)>;
+
+  static inline HistoricalTxStatus is_tx_committed_v2(
+    kv::Consensus* consensus,
+    ccf::View view,
+    ccf::SeqNo seqno,
+    std::string& error_reason)
+  {
+    if (consensus == nullptr)
+    {
+      error_reason = "Node is not fully configured";
+      return HistoricalTxStatus::Error;
+    }
+
+    const auto tx_view = consensus->get_view(seqno);
+    const auto committed_seqno = consensus->get_committed_seqno();
+    const auto committed_view = consensus->get_view(committed_seqno);
+
+    const auto tx_status = ccf::evaluate_tx_status(
+      view, seqno, tx_view, committed_view, committed_seqno);
+    switch (tx_status)
+    {
+      case ccf::TxStatus::Unknown:
+      case ccf::TxStatus::Pending:
+        error_reason = fmt::format(
+          "Only committed transactions can be queried. Transaction {}.{} "
+          "is "
+          "{}",
+          view,
+          seqno,
+          ccf::tx_status_to_str(tx_status));
+        return HistoricalTxStatus::PendingOrUnknown;
+      case ccf::TxStatus::Invalid:
+        error_reason = fmt::format(
+          "Only committed transactions can be queried. Transaction {}.{} "
+          "is "
+          "{}",
+          view,
+          seqno,
+          ccf::tx_status_to_str(tx_status));
+        return HistoricalTxStatus::Invalid;
+      case ccf::TxStatus::Committed:;
+    }
+
+    return HistoricalTxStatus::Valid;
+  };
+
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunused-function"
 
+  static ccf::endpoints::EndpointFunction adapter_v2(
+    const HandleHistoricalQuery& f,
+    AbstractStateCache& state_cache,
+    const CheckHistoricalTxStatus& available,
+    const TxIDExtractor& extractor = txid_from_header)
+  {
+    return [f, &state_cache, available, extractor](
+             endpoints::EndpointContext& args) {
+      // Extract the requested transaction ID
+      ccf::TxID target_tx_id;
+      {
+        const auto tx_id_opt = extractor(args);
+        if (tx_id_opt.has_value())
+        {
+          target_tx_id = tx_id_opt.value();
+        }
+        else
+        {
+          return;
```

Comment:
I know it's the same as in v1, but... what happens exactly in this case?

Reply:
That's a good question, whatever we do when we return from a handler without setting anything, which appears to be an empty 200?

Reply:
So no, what happens is the extractor has already set the return code and the error body.